cql3/statements/create_index_statement.cc

```diff
@@ -131,7 +131,7 @@ std::vector<::shared_ptr<index_target>> create_index_statement::validate_while_e
         }
 
         // Origin TODO: we could lift that limitation
-        if ((schema->is_dense() || !schema->thrift().has_compound_comparator()) && cd->is_primary_key()) {
+        if ((schema->is_dense() || !schema->is_compound()) && cd->is_primary_key()) {
```

Comment:
Is this the right change? Since we don't have thrift any more, what does thrift().has_compound_comparator() mean and why is the same as is_compound (which we still have)?

Reply:
i intended to preserve the existing behavior and not to break it. `is_compound()` is equivalent to `schema->thrift().has_compound_comparator())` . but yeah, will take a closer look at it. trying to be a good archeologist ..